base/expr.jl

```diff
@@ -435,3 +435,32 @@ macro generated(f)
         error("invalid syntax; @generated must be used with a function definition")
     end
 end
+
+
+"""
+    @atomic ex
+
+Mark `ex` as being performed atomically.
+
+```julia
+mutable struct Atomic{T}; @atomic x::T; end
+a = Atomic(1)
+@atomic a.x = 2 # set field x of a
+@atomic a.x # fetch field x or a
+# TODO: @atomic a.x += 1 # increment field x of a
+# TODO: @atomic +!(a.x, 1) # increment field x of a
+# TODO: @atomic a.x, z = y, a.x # swap field x of a with y and put the old value in z
```

Comment:
These should probably be removed or put under a heading like "the following are planned but not implemented yet:".